<launch>
    <arg name="tmap" default="topological_map" />
    <arg name="generate_topo_map" default="false" />

    <!-- Thorvald Launch -->
    <include file="$(find uol_cmp9767m_base)/launch/thorvald-sim.launch" >
        <arg name="second_robot" value="false"/>
        <arg name="obstacles" value="false"/>
        <arg name="fake_localisation" value="true"/>
        <arg name="map_server" value="true"/> 
    </include>

    <!-- Topological navigation -->
    <include file="topological_navigation.launch" >
        <arg name="tmap" value="$(arg tmap)" />
    </include>

    <!-- generate topo map -->
    <group if="$(arg generate_topo_map)">
        <node pkg="thorvald_2dnav" type="generate_topo_map.py" name="generate_topo_map" output="screen"/>
        <node pkg="topological_utils" type="load_yaml_map.py" name="load_yaml_map" args="$(find thorvald_2dnav)/maps/empty_map.yaml -f --pointset $(arg tmap)"  />
    </group>
   
    <!-- weeding nodes -->
    <group unless="$(arg generate_topo_map)">
        <node pkg="vision" type="vision_node.py" name="vision_node" args="thorvald_001" output="screen"/>
        <node pkg="state_machine" type="state_machine.py" name="state_machine" args="thorvald_001" output="screen"/>
        <node pkg="thorvald_2dnav" type="translation_move.py" name="translation_move_node" args="thorvald_001" output="screen"/>
    </group>

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find thorvald_2dnav)/thorvald_rviz_config.rviz" />

    <!--- Run AMCL --> 
    <include file="$(find amcl)/examples/amcl_omni.launch" />
</launch>